cmake/kokkos_cxx_std.cmake

```
@@ -0,0 +1,211 @@
+KOKKOS_CFG_DEPENDS(CXX_STD COMPILER_ID)
+
+FUNCTION(kokkos_set_cxx_standard_feature standard)
+  SET(EXTENSION_NAME CMAKE_CXX${standard}_EXTENSION_COMPILE_OPTION)
+  SET(STANDARD_NAME  CMAKE_CXX${standard}_STANDARD_COMPILE_OPTION)
+  SET(FEATURE_NAME   cxx_std_${standard})
+  #message("HAVE ${FEATURE_NAME} ${${EXTENSION_NAME}} ${${STANDARD_NAME}}")
+  #CMake's way of telling us that the standard (or extension)
+  #flags are supported is the extension/standard variables
+  IF (NOT DEFINED CMAKE_CXX_EXTENSIONS)
+    IF(KOKKOS_DONT_ALLOW_EXTENSIONS)
+      GLOBAL_SET(KOKKOS_USE_CXX_EXTENSIONS OFF)
+    ELSE()
+      GLOBAL_SET(KOKKOS_USE_CXX_EXTENSIONS ON)
+    ENDIF()
+  ELSEIF(CMAKE_CXX_EXTENSIONS)
+    IF(KOKKOS_DONT_ALLOW_EXTENSIONS)
+      MESSAGE(FATAL_ERROR "The chosen configuration does not support CXX extensions flags: ${KOKKOS_DONT_ALLOW_EXTENSIONS}. Must set CMAKE_CXX_EXTENSIONS=OFF to continue") 
+    ELSE()
+      GLOBAL_SET(KOKKOS_USE_CXX_EXTENSIONS ON)
+    ENDIF()
+  ELSE()
+    GLOBAL_SET(KOKKOS_USE_CXX_EXTENSIONS OFF)
+  ENDIF()
+
+  IF (KOKKOS_USE_CXX_EXTENSIONS AND ${EXTENSION_NAME})
+    MESSAGE(STATUS "Using ${${EXTENSION_NAME}} for C++${standard} extensions as feature")
+    GLOBAL_SET(KOKKOS_CXX_STANDARD_FEATURE ${FEATURE_NAME})
+  ELSEIF(NOT KOKKOS_USE_CXX_EXTENSIONS AND ${STANDARD_NAME})
+    MESSAGE(STATUS "Using ${${STANDARD_NAME}} for C++${standard} standard as feature")
+    GLOBAL_SET(KOKKOS_CXX_STANDARD_FEATURE ${FEATURE_NAME})
+  ELSE()
+    #nope, we can't do anything here
+    MESSAGE(STATUS "C++${standard} is not supported as a compiler feature - choosing custom flags")
+    GLOBAL_SET(KOKKOS_CXX_STANDARD_FEATURE "")
+  ENDIF()
+
+  IF(NOT ${FEATURE_NAME} IN_LIST CMAKE_CXX_COMPILE_FEATURES)
+    IF (KOKKOS_CXX_COMPILER_ID STREQUAL "NVIDIA")
+      MESSAGE(STATUS "nvcc_wrapper does not support TARGET_COMPILE_FEATURES")
+      GLOBAL_SET(KOKKOS_CXX_STANDARD_FEATURE "")
+    ELSE()
+      MESSAGE(FATAL_ERROR "Compiler ${KOKKOS_CXX_COMPILER_ID} should support ${FEATURE_NAME}, but CMake reports feature not supported")
+    ENDIF()
+  ENDIF()
+ENDFUNCTION(kokkos_set_cxx_standard_feature)
+
+
+
+
+# From CMake 3.10 documentation
+#CMake is currently aware of the C++ standards compiler features
+
+#AppleClang: Apple Clang for Xcode versions 4.4 though 6.2.
+#Clang: Clang compiler versions 2.9 through 3.4.
+#GNU: GNU compiler versions 4.4 through 5.0.
+#MSVC: Microsoft Visual Studio versions 2010 through 2017.
+#SunPro: Oracle SolarisStudio versions 12.4 through 12.5.
+#Intel: Intel compiler versions 12.1 through 17.0.
+#
+#Cray: Cray Compiler Environment version 8.1 through 8.5.8.
+#PGI: PGI version 12.10 through 17.5.
+#XL: IBM XL version 10.1 through 13.1.5.
+
+#This can run at any time
+KOKKOS_OPTION(CXX_STANDARD "" STRING "The C++ standard for Kokkos to use: c++11, c++14, or c++17")
```

```suggestion
KOKKOS_OPTION(CXX_STANDARD "" STRING "The C++ standard for Kokkos to use: 11, 14, 17, 1Y, 1Z, 2A")
```
This should match the code below.

---

Fixed.